spec/integration/apply_spec.rb

```diff
@@ -289,7 +289,7 @@ def task_plugin_inventory
     context "with a puppet_agent installed" do
       before(:all) do
         # Deferred must use puppet >= 6
-        target = 'puppet_6'
+        target = 'ubuntu_node'
```

Comment:
I think we actually want to install the agent on the `puppet_6` node and verify against that target. Using the `agent_version_inventory` it should pick up the correct config.

Reply:
I think we want to set `uri` to the puppet_6 node. I kind of think the easiest way to do that is to at the top have:
```
def puppet_6
  @puppet_6 ||= docker_inventory(root: true).get_target('puppet_6_node')
end
```
and then set the uri to `puppet_6` the 'with puppet_agent' context. But you could also just add 'puppet_6' and puppet_5 functions to the conn helper that do the same thing.